### PR TITLE
fix: disable DevtoolPlugin when options.devtool=false

### PR DESCRIPTION
**Pull request by LingyuCoder**
## Summary

Disable DevtoolPlugin when options.devtool=false

Can not simplely use devtool.enabled() because it works for module loader.
Method add_module() will also works when devtool is an empty string or been some other illegal strings.
So return directly and add a new method plugin_enabled() to disable the DevtoolPlugin.

### CODE REVIEW

**Comment by hardfist** [comment]
!bench

**Review by hyf0 — DISMISSED** [review]

**Comment by ahabhgk** [comment]
snapshot needs update, hash changed is caused by https://github.com/web-infra-dev/rspack/blob/eb22d900e18439afe29627e5bb7dcea378f6bcc5/crates/rspack_plugin_devtool/src/lib.rs#L95

File: crates/rspack_binding_options/src/options/mod.rs [hunk 1]
```diff
@@ -222,17 +222,20 @@ impl RawOptionsApply for RawOptions {
...
-        append: !devtool.hidden(),
-        namespace: output.unique_name.clone(),
-        columns: !devtool.cheap(),
-        no_sources: devtool.no_sources(),
-        public_path: None,
-      })
-      .boxed(),
-    );
+
+    if devtool.plugin_enabled() {
```
Comment (ahabhgk) [inline crates/rspack_binding_options/src/options/mod.rs:226]: `devtool.source_map()` looks better to me

Comment (ahabhgk) [reply crates/rspack_binding_options/src/options/mod.rs:226]: I didn't implement `'eval'` so don't need to care about `'eval'`, even if I need to implement it in the future I will add an another plugin to do that

Comment (LingyuCoder) [reply crates/rspack_binding_options/src/options/mod.rs:226]: > `devtool.source_map()` looks better to me

source_map() has been exists and not works for devtool=eval

**Review by ahabhgk — APPROVED** [review]

**Pull request merged**